src/presenter.rs

```diff
@@ -81,21 +81,63 @@ impl Presenter {
             self.print_vulnerability(vulnerability, &tree);
         }
 
+        let deny_unmaintained = self
+            .config
+            .deny_warnings
+            .contains(&WarningKind::Unmaintained);
+        let deny_yanked = self.config.deny_warnings.contains(&WarningKind::Yanked);
+
+        //
+        //  could instead do something like this (if Warning implemented Copy):
+        //
+        //  let (denied, others): (Vec<Warning>, Vec<Warning>) =
+        //     report.warnings.iter().partition(|&w| match w.kind {
+        //         warning::Kind::Unmaintained { .. } => deny_unmaintained,
+        //         warning::Kind::Yanked { .. } => deny_yanked,
+        //         _ => false,
+        //     });
+        let mut num_denied = 0;
+        let mut num_not_denied = 0;
+        for warning in &report.warnings {
+            match warning.kind {
+                warning::Kind::Unmaintained { .. } => {
+                    if deny_unmaintained {
+                        num_denied += 1
+                    } else {
+                        num_not_denied += 1
+                    }
+                }
+                warning::Kind::Yanked { .. } => {
+                    if deny_yanked {
+                        num_denied += 1
+                    } else {
+                        num_not_denied += 1
+                    }
+                }
+                _ => num_not_denied += 1,
+            }
+        }
+
         if !report.warnings.is_empty() {
             println!();
 
-            let warning_word = if report.warnings.len() != 1 {
+            let warning_word = if num_denied != 1 {
                 "warnings"
             } else {
                 "warning"
             };
 
-            if self.config.deny_warnings {
-                status_err!("{} {} found", report.warnings.len(), warning_word);
+            status_err!("{} denied {} found", num_denied, warning_word);
```

Comment:
Some decisions to be made about how to display this...

Reply:
Generally I've looked to `rustc` and `clippy` for ideas and conventions for this sort of thing.

They promote `warning:` -> `error:` in this case, and add a `note:` saying that deny warnings has been enabled.